docs/src/appendix/scarb-toml.md

```diff
@@ -0,0 +1,236 @@
+```toml
+[profile.dev.cairo]
+unstable-add-statements-functions-debug-info = true
+```
+
+#### `inlining-strategy`
+This field is responsible for setting the inlining strategy to be used by compiler when building the package. The possible values are `default` or `avoid`. If avoid strategy is set, the compiler will only inline function annotated with `#[inline(always)]` attribute.
+```toml
+[profile.dev.cairo]
+inlining-strategy = "avoid"
+```
+
+#### Example of configuration which allows [coverage](https://foundry-rs.github.io/starknet-foundry/testing/coverage.html) report generation
+```toml
+[profile.dev.cairo]
+unstable-add-statements-code-locations-debug-info = true
+unstable-add-statements-functions-debug-info = true
+inlining-strategy = "avoid"
+```
+
+### `[features]`
+A package defines a set of named features in the `[features]` section of `Scarb.toml` file. Each defined feature can list other features that should be enabled with it. All fields are optional.
+```toml
+[features]
+# ...
+```
+
+#### `<feature-name>`
+The `<feature-name>` field specifies the name of the feature and list of other features that should be enabled with it.
+```toml
+[features]
+enable_for_tests = []
+```
+
+#### Example of `Scarb.toml` allowing conditional contracts compilation
+Firstly, define a contract in the src directory with a `#[cfg(feature: '<FEATURE_NAME>')]` attribute:
+```rust
+#[starknet::contract]
+#[cfg(feature: 'enable_for_tests')]
+mod MockContract {
+    // ...
+}
+```
+
+The update Scarb.toml so it includes the following lines:
+```toml
+[features]
+enable_for_tests = []
+```
+
+### `[[target.starknet-contract]]`
+Scarb supports registering targets that are handled by Scarb extensions. Such targets are called external. All fields are optional.
+```toml
+[[target.starknet-contract]]
+# ...
+```
+
+#### `sierra`
+The `sierra` fields specifies whether Sierra codegen should be enabled.
+```toml
+[[target.starknet-contract]]
+sierra = true
+```
+
+#### `casm`
+The `casm` fields specifies whether Casm codegen should be enabled.
+```toml
+[[target.starknet-contract]]
+casm = true
+```
+
+
+#### `build-external-contracts`
+The `build-external-contracts` accepts a list of strings, each of which is a reference to a contract defined in a dependency. The package that implements this contracts need to be declared as a dependency of the project in `[dependencies]`.
+```toml
+[[target.starknet-contract]]
+build-external-contracts = ["openzeppelin::account::account::Account"]
+```
+
+#### Example of configuration which allows to use external contracts in tests
+```toml
+# ...
+[dependencies]
+starknet = ">=2.8.2"
+openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", branch = "cairo-2" }
+
+[[target.starknet-contract]]
+build-external-contracts = ["openzeppelin::account::account::Account"]
+# ...
+```
```

Comment:
While I get why this was included, this is kind of duplicating Scarb docs and adds an additional burden for us of maintaining it. 

Maybe let's just say that we handle `[features] and link to both a section in our docs that explain conditional compilation and to scarb docs? Similar on how scarb does it in their reference

<img width="831" alt="image" src="https://github.com/user-attachments/assets/107ac8d9-2e50-4616-8c64-d75983e5a71a">

Same goes for starknet contract target and maybe even for `unstable-...`.